### ISSUE TITLE
RRTMGP memory leak

### ISSUE THREAD

**Issue by whannah1** [E3SM-Project/E3SM#4532]
During initial Perlmutter testing with the MMF @ndkeen reported a consistent "out of memory" error when trying to run the MMF for 5 days with 16 rad columns. These tests have only used GNU and the problem is dependent on the version. GNU 8 does not exhibit this problem, whereas 9 and 10 do. 

I was running a different MMF configuration with 4 radiation columns and was not seeing the error, but I quickly reproduced the problem when I used 64 rad columns. So the problem is definitely in the radiation. The RRTMG interface for the MMF is currently broken, so I haven't been able to verify is it is also affected.

I spent some time exploring the idea that the problem was related to the treatment of allocated pointers that may not be deallocated when they fall out of scope. I tried to find all instances of this and convert the variables to allocatable, but it didn't change the outcome. 

@brhillman  suggested that this may be a bug that has been fixed in the main RRTMGP repo, which hasn't been been pulled into our fork. Ben updated the fork in a branch, but we have yet to verify if this branch also exhibits the memory leak.

**Comment by ndkeen** [comment]
On cori-knl, using the branch `ndk/machinefiles/cori-test-gnu9` I can use --compiler=gnu, --compiler=gnu9 or --compiler=gnu10. I ran the following 4 cases:
```
SMS_P96.ne4pg2_ne4pg2.F-MMFXX.cori-knl_gnu.20210904_105521_xpe8y1
SMS_P96.ne4pg2_ne4pg2.F-MMFXX.cori-knl_gnu10.20210904_135315_ps5v7q
SMS_P96.ne4pg2_ne4pg2.F-MMFXX.cori-knl_gnu9.20210904_105521_xpe8y1
SMS_P96.ne4pg2_ne4pg2.F-MMFXX.cori-knl_intel.20210904_110405_70wx0d
```
where our default gnu/intel complete 5 days (gnu needs about 80 minutes and intel needs 60 min), however both gnu9 and gnu10 fail with OOM (after about 18 min). The failing cases complete about 80 timesteps. We verified the memory growth happens at each radiation call.

On Perlmutter, we only have gnu9/gnu10.

And using `brhillman/mmf/integrate-rrtmgpxx-new` the memory issue is not present.
